source/MapDetailPanel.cpp

```diff
@@ -442,7 +442,7 @@ void MapDetailPanel::DrawInfo()
 				// The same "planet" may appear multiple times in one system,
 				// providing multiple landing and departure points (e.g. ringworlds).
 				const Planet *planet = object.GetPlanet();
-				if(planet->IsWormhole() || !planet->IsAccessible(player.Flagship()) || shown.count(planet))
+				if(planet->IsWormhole() || shown.count(planet))
```

Comment:
You probably need to gate accessibility on having visited it here - otherwise the player is told "hey, look, planet! (but you can't go there!)" before visiting it.